### PR TITLE
fix(parser): small files no longer silently dropped from indexing

### PR DESCRIPTION
**Pull request by alfhen**
## Summary

Fixes a trust-class bug found during PR #772's dogfood (2026-07-16): a ~5-line test file containing only a bare `test('...', () => {...})` call was **silently dropped from indexing entirely** — no error, no chunk, absent from the manifest.

## Root cause

`chunkByAST` (`packages/parser/src/ast/chunker.ts`) recognizes functions, classes, interfaces, and const/let/var declarations as top-level chunks. Anything else at the top level (bare call expressions like `test(...)`, `it(...)`, `describe(...)`, plain statements, etc.) falls through to an "uncovered code" fallback that covers whatever wasn't recognized.

That fallback is filtered by `minChunkSize` (7 lines, in production — `Math.floor(chunkSize / 10)` with the default `chunkSize` of 75) — a guard meant to suppress noise chunks for small leftover gaps *alongside* real function/class chunks in an otherwise normal file (e.g. a lone blank-line gap between two functions).

The bug: when a file has **zero** recognized top-level nodes, `coveredRanges` is empty, so the single "uncovered" range spans the *entire file*. Applying the same size guard there doesn't shrink a noise chunk — it deletes the file's only chance at representation. A file under the threshold (e.g. 5 real lines) produced zero chunks, no manifest entry, and no error.

This is exactly analogous to a previously-fixed bug: barrel/re-export-only files (`export { foo } from './foo'`) got the same treatment and were fixed by bypassing `minChunkSize` when `fileExports.length > 0` (see the "barrel/re-export files" tests in `chunker.test.ts`). The zero-top-level-node case was the same class of bug, left unfixed.

## Repro (verbatim)

```ts
// src/tiny.test.ts (5 lines)
import { test, expect } from 'vitest';

test('does something', () => {
  expect(1).toBe(1);
});
```

Indexing this file before the fix:

```
manifest.json: "files": {}
structural.db: SELECT count(*) FROM chunks;  -> 0
```

No error was printed. Binary-searching the boundary (padding with blank lines) showed the exact cliff: a file whose content (via `content.split('\n')`, including the trailing-newline artifact) has **6 lines pass, 5 lines (or fewer) vanish** — i.e. `lines.length >= minChunkSize (7)`.

Confirmed via cases:

| shape | before fix | after fix |
|---|---|---|
| 5-line bare `test()` call | 0 chunks, absent from manifest | 1 chunk |
| single-function file (3 lines) | 1 chunk (unaffected — already a recognized top-level node) | 1 chunk |
| 1-line `export const foo = 1;` | 1 chunk (unaffected — `lexical_declaration` is always a top-level node regardless of size) | 1 chunk |
| empty file | 0 chunks | 0 chunks (unchanged — correctly still skipped) |
| whitespace-only file | 0 chunks | 0 chunks (unchanged — correctly still skipped) |

## Fix

`extractUncoveredCode` now also bypasses `minChunkSize` when the file produced no other (top-level) chunks — the same bypass condition already used for `hasExports` (barrel files), just widened to cover this case. The `chunk.content.length > 0` check still applies either way, so empty/whitespace-only files are correctly unaffected.

```ts
const hasExports = fileExports && fileExports.length > 0;
const skipMinSize = hasExports || fileHasNoTopLevelChunks;

return uncoveredRanges
  .map(range => createChunkFromRange(...))
  .filter(chunk => (skipMinSize ? chunk.content.length > 0 : isValidChunk(chunk, minChunkSize)));
```

`fileHasNoTopLevelChunks` = `topLevelNodes.length === 0`, threaded in from `chunkByAST`. Since `coveredRanges` is derived 1:1 from `topLevelNodes`, this is always exactly the "whole file is one uncovered range" case — no ambiguity with the multi-gap case the guard was designed for.

## Dogfood evidence (re-running #772's failing case)

Fixture: `src/util.ts` (exports `add`) + `src/util.test.ts` (5 lines, imports `add`, one bare `test(...)` call):

```
manifest.json:
  "src/util.test.ts": { "chunkCount": 1, ... }
  "src/util.ts":      { "chunkCount": 1, ... }

structural.db chunks:
  src/util.test.ts|1|6|block
  src/util.ts|1|3|function
```

`findTestAssociationsFromChunks(["src/util.ts"], chunks, ...)` now correctly returns:

```json
{ "src/util.ts": ["src/util.test.ts"] }
```

Previously this was silently empty, since `util.test.ts` never made it into `chunks` at all.

## Impact note

This changes what enters `repoChunks` for very small files (mainly tiny standalone test files). Freshly captured/regenerated harness fixtures may pick up small files they previously didn't. **This PR is sequenced before any harness/corpus recalibration sweep**, so certified corpora should be captured post-fix.

## Testing

- New regression tests in `packages/parser/src/ast/chunker.test.ts` (`chunkByAST` boundary, explicit `minChunkSize: 7` matching production) and `packages/parser/src/chunker.test.ts` (`chunkFile` entry point, default options): bare `test()` call, single-function file, 1-line export, empty file (still skipped), whitespace-only file (still skipped).
- Full test suite: `npm test` — all packages green (parser 1062, core 35 files, review 47 files, cli 832, action 39).
- `npm run test:e2e:typescript -w packages/cli` (Zod, 443 files / 4786 chunks) — clean index, no regressions.
- `lien delta` — exit 0 (2 "worsened" notices on `chunkByAST`/`extractUncoveredCode`, no new threshold crossings).
- `npm run typecheck`, `npm run lint`, `npm run format:check`, `npm run build` — all clean.

## Test plan

- [x] Reproduced the silent-drop bug in an isolated fixture (verbatim above)
- [x] Root-caused to `isValidChunk`/`minChunkSize` applied to the whole-file fallback chunk
- [x] Minimal fix + regression tests (parser + chunker levels)
- [x] Full test suite green
- [x] TypeScript e2e green on a real 443-file repo
- [x] `lien delta` exit 0
- [x] Dogfooded the exact #772 failing shape end-to-end (chunk + test-association)

---

### Lien Review

✅ **Good** - No complexity issues found.

> [!NOTE]
> **Low Risk**
>
> This PR fixes a narrow, well-defined indexing bug: tiny files containing only bare top-level calls (e.g., a 5-line vitest test) were silently dropped because the uncovered-code fallback applied the same minChunkSize guard meant for inter-chunk gaps. The fix bypasses minChunkSize when a file has zero recognized top-level nodes, while preserving the empty/whitespace-only filter via chunk.content.length > 0. The change is internal to the parser's AST chunker, adds no public API surface, and is covered by regression tests at both the AST boundary and the chunkFile entry point. Blast radius is limited to indexing paths.
>
> - extractUncoveredCode now skips minChunkSize when topLevelNodes.length === 0, mirroring the existing barrel-file bypass
> - Added regression tests for the 5-line bare test() case, single-function file, 1-line export, empty file, and whitespace-only file
> - No public API or export surface changes; only indexing behavior for very small non-declaration files changes

<sup>Reviewed by [Lien Review](https://lien.dev) for commit 2b79fdb. Updates automatically on new commits.</sup>

### CODE REVIEW

File: .changeset/parser-min-chunk-drop.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+'@liendev/parser': patch
+'@liendev/core': patch
+---
+
```
[the remark this answers is not in the document]
Comment (alfhen) [reply .changeset/parser-min-chunk-drop.md:5]: Fixed in 2b79fdb7 — reworded to: `chunkByAST`'s own default is 5, and production always calls it via `chunkFile`, which computes 7 from `Math.floor(chunkSize / 10)`. Thanks for catching the imprecise attribution.

File: .changeset/parser-min-chunk-drop.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Fix a silent indexing gap: small files with no top-level function/class/interface/declaration (e.g. a single bare `test(...)` call) could be dropped from the index entirely — no chunk, no manifest entry, no error.
+
```
[the remark this answers is not in the document]
Comment (alfhen) [reply .changeset/parser-min-chunk-drop.md:7]: Re-verified: `extractUncoveredCode`'s filter is `skipMinSize ? chunk.content.length > 0 : isValidChunk(...)` — both branches of the ternary require `chunk.content.length > 0`, and `createChunkFromRange` sets `content` via `.trim()` on the joined lines, so an empty or whitespace-only file always produces `content === ''` regardless of which branch is taken. This is also directly asserted by the new regression tests (`should still produce no chunks for an empty file` / `...for a whitespace-only file` in both `chunker.test.ts` files), which pass in CI. Dismissing — claim confirmed correct, not a bug.
